crates/pallets/starknet/src/runtime_api.rs

```diff
@@ -32,42 +32,6 @@ pub enum StarknetTransactionExecutionError {
 }
 
 sp_api::decl_runtime_apis! {
-    pub trait StarknetRuntimeApi {
-        /// Returns the nonce associated with the given address in the given block
-        fn nonce(contract_address: ContractAddress) -> Nonce;
-        /// Returns a storage slot value
-        fn get_storage_at(address: ContractAddress, key: StorageKey) -> Result<StarkFelt, DispatchError>;
-        /// Returns a `Call` response.
-        fn call(address: ContractAddress, function_selector: EntryPointSelector, calldata: Calldata) -> Result<Vec<Felt252Wrapper>, DispatchError>;
-        /// Returns the contract class hash at the given address.
-        fn contract_class_hash_by_address(address: ContractAddress) -> ClassHash;
-        /// Returns the contract class for the given class hash.
-        fn contract_class_by_class_hash(class_hash: ClassHash) -> Option<ContractClass>;
-        /// Returns the chain id.
-        fn chain_id() -> Felt252Wrapper;
-        /// Returns fee estimate
-        fn estimate_fee(transactions: Vec<UserTransaction>) -> Result<Vec<(u64, u64)>, DispatchError>;
-        /// Filters extrinsic transactions to return only Starknet transactions
-        ///
-        /// To support runtime upgrades, the client must be unaware of the specific extrinsic
-        /// details. To achieve this, the client uses an OpaqueExtrinsic type to represent and
-        /// manipulate extrinsics. However, the client cannot decode and filter extrinsics due to
-        /// this limitation. The solution is to offload decoding and filtering to the RuntimeApi in
-        /// the runtime itself, accomplished through the extrinsic_filter method. This enables the
-        /// client to operate seamlessly while abstracting the extrinsic complexity.
-        fn extrinsic_filter(xts: Vec<<Block as BlockT>::Extrinsic>) -> Vec<Transaction>;
-        fn get_events_for_tx_hash(xts: Vec<<Block as BlockT>::Extrinsic>, chain_id: Felt252Wrapper, tx_hash: Felt252Wrapper) -> Option<(TxType, Vec<StarknetEvent>)>;
-
-        /// Return the list of StarknetEvent evmitted during this block, along with the hash of the starknet transaction they bellong to
-        ///
-        /// `block_extrinsics` is the list of all the extrinsic executed during this block, it is used in order to match
-        fn get_starknet_events_and_their_associated_tx_hash(block_extrinsics: Vec<<Block as BlockT>::Extrinsic>, chain_id: Felt252Wrapper) -> Vec<(Felt252Wrapper, StarknetEvent)>;
-        /// Return the outcome of the tx execution
-        fn get_tx_execution_outcome(tx_hash: TransactionHash) -> Option<Vec<u8>>;
-        /// Return the block context
-        fn get_block_context() -> BlockContext;
-    }
-
     pub trait ConvertTransactionRuntimeApi {
```

Comment:
Move `ConvertTransactionRuntimeApi` too.

Basically, move the whole file. Entierly